iml-services/iml-ntp/src/main.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+#![feature(async_await)]
+
+mod error;
+
+use error::NtpError;
+use futures::{
+    compat::{Future01CompatExt, Stream01CompatExt},
+    future::{FutureExt, TryFuture, TryFutureExt},
+    stream::{Stream, StreamExt, TryStreamExt},
+};
+use iml_postgres::connect;
+use iml_rabbit::connect_to_rabbit;
+use iml_service_queue::service_queue::{consume_service_queue, data_only, into_deserialized};
+use iml_wire_types::{
+    ntp::{TimeStatus, TimeSync},
+    Fqdn,
+};
+use std::io;
+use tokio_postgres::Client;
+use tracing_subscriber::{fmt::Subscriber, EnvFilter};
+
+/// This async function will retrieve the content type id of the specified model. A future
+/// containing the content type id of the model is returned.
+async fn get_managed_host_content_type_id(client: &mut Client) -> Result<i32, NtpError> {
+    client
+        .prepare("SELECT id FROM django_content_type WHERE model='managedhost';")
+        .compat()
+        .map_ok(move |statement| client.query(&statement, &[]).compat())
+        .try_flatten_stream()
+        .try_next()
+        .await?
+        .map(|row| row.get::<_, i32>(0))
+        .ok_or_else(|| NtpError::IoError(io::Error::new(io::ErrorKind::NotFound, "managedhost")))
+}
+
+/// This async function will retrieve the managed host id for a given fqdn. A future
+/// containing the managed host id is returned.
+async fn get_managed_host_id(fqdn: &str, client: &mut Client) -> Result<i32, NtpError> {
+    client
+        .prepare("SELECT id FROM chroma_core_managedhost WHERE fqdn=$1;")
+        .compat()
+        .map_ok(move |statement| client.query(&statement, &[&fqdn]).compat())
+        .try_flatten_stream()
+        .try_next()
+        .await?
+        .map(|row| row.get::<_, i32>(0))
+        .ok_or_else(|| {
+            NtpError::IoError(io::Error::new(
+                io::ErrorKind::NotFound,
+                "Error getting host id",
+            ))
+        })
+}
+
+async fn get_active_alert_for_fqdn(fqdn: &str, client: &mut Client) -> Result<i32, NtpError> {
+    client
+        .prepare(r#"SELECT S.id FROM chroma_core_alertstate AS S LEFT JOIN chroma_core_managedhost AS MH 
```

Comment:
Use  [`Name` traits](https://docs.rs/iml-wire-types/0.2.0/iml_wire_types/db/trait.Name.html)  for the table names.